substrate/frame/support/test/tests/pallet.rs

```diff
@@ -282,6 +287,7 @@ pub mod pallet {
 		/// error doc comment put in metadata
 		InsufficientProposersBalance,
 		#[deprecated = "test"]
+		#[allow(deprecated)]
```

Comment:
Why do I need to put here `deprecated` and `allow(deprecated)`?

Reply:
`deprecated` is for generating deprecation info in metadata-ir. `allow(deprecated)` is to silence the warnings from generated code

Reply:
`allow(deprecated)` is no longer needed now, done as you suggested for generated code.